File: app/src/main/java/com/codebosses/comical/activities/SplashActivity.java
package com.codebosses.comical.activities;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import com.codebosses.comical.R;
import com.codebosses.comical.endpoints.EndpointKeys;
import com.codebosses.comical.utils.PrefUtils;

public class SplashActivity extends AppCompatActivity {

    private Handler handler;
    private Context context;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        context = this;

        handler = new Handler();
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {

                if ((Boolean) PrefUtils.getFromPrefs(context, EndpointKeys.IS_USER_FIRST_TIME, true)) {
                    Intent intent = new Intent(context, IntroActivity.class);
                    startActivity(intent);
                    finish();
                } else {
                    Intent intent = new Intent(context, MainActivity.class);
                    startActivity(intent);
                    finish();
                }

            }
        }, 3000);

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (handler != null) {
            handler.removeCallbacks(null);
        }
    }
}

File: app/src/main/java/com/codebosses/comical/ui/splash/SplashActivity.kt
package com.codebosses.comical.ui.splash

import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import com.codebosses.comical.R
import com.codebosses.comical.ui.intro.IntroActivity
import com.codebosses.comical.utils.PrefUtils
import com.codebosses.comical.utils.extensions.startActivityWithFinish
import com.codebosses.comical.ui.main.MainActivity
import com.codebosses.comical.utils.LogUtil

class SplashActivity : AppCompatActivity() {

    private var handler: Handler? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        handler = Handler()

        LogUtil.debug("notification",PrefUtils.deviceToken)

        handler?.postDelayed({ ->
            startActivityWithFinish(if(PrefUtils.firstRun){
                IntroActivity::class.java
            }
            else{
                MainActivity::class.java
            })
        },3000)
    }

    override fun onDestroy() {
        super.onDestroy()
        if(handler != null){
            handler?.removeCallbacks(null)
        }
    }
}
